features/detect_nproc.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+#
+# Exit with a status reflecting the CPU count if we can determine it. If not
+# assume two CPUs since at least one test, API/trehash.c, deadlocks if the CPU
+# count is one.
+if getconf _NPROCESSORS_ONLN >/dev/null 2>&1
+then
+	# Some systems have a getconf that supports this var. Use it in preference
+	# to the options below because those other mechanisms include offline CPUs.
+	exit $(getconf _NPROCESSORS_ONLN)
```

Comment:
This should work unless we are running on supercomputers with greater than 255 cores. And historically `ksh` has run on supercomputers.

Reply:
Good point. Fixed to ensure we never exit with less than 2 (since at least one test fails to run if nproc is 1) or more than 64 (which is the default for most tests if N_PROC isn't defined).